test/e2e/fixture/fixture.go

```diff
@@ -136,22 +133,15 @@ func EnsureCleanState() {
 
 	start := time.Now()
 
+	policy := v1.DeletePropagationBackground
 	// delete resources
-	text, err := Run("", "kubectl", "get", "app", "-o", "name")
-	CheckError(err)
-	for _, name := range strings.Split(text, "\n") {
-		if name != "" {
-			appName := strings.TrimPrefix(name, "application.argoproj.io/")
-			// we cannot delete any app, if an op is in progress
-			_, _ = RunCli("app", "terminate-op", appName)
-			// is it much more reliable to get argocd to delete an app than kubectl, deleting directly can make ArgoCD stuck
-			_, _ = RunCli("app", "delete", appName)
-		}
-	}
-	FailOnErr(Run("", "kubectl", "delete", "appprojects", "--field-selector", "metadata.name!=default"))
-	// takes around 5s, so we don't wait
+	CheckError(AppClientset.ArgoprojV1alpha1().Applications(ArgoCDNamespace).DeleteCollection(&v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{}))
+	CheckError(AppClientset.ArgoprojV1alpha1().AppProjects(ArgoCDNamespace).DeleteCollection(
+		&v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{FieldSelector: "metadata.name!=default"}))
+	CheckError(KubeClientset.CoreV1().Secrets(ArgoCDNamespace).DeleteCollection(
+		&v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{LabelSelector: testingLabel + "=true"}))
+
 	FailOnErr(Run("", "kubectl", "delete", "ns", "-l", testingLabel+"=true", "--field-selector", "status.phase=Active", "--wait=false"))
```

Comment:
Do you need to delete this line too?

Reply:
We still need to deleted temporal namespaces. There is no API to delete collection of namespaces, so I kept one `kubectl delete ...` . Would you prefer to use api here for consistency?

Reply:
Nope, not fussed.

Reply:
Hmmm.. How can there be no API for this, I'd assumed that kubectl just makes API calls?

Reply:
you are right. there is API, but no API to delete collection, only one by one deletion